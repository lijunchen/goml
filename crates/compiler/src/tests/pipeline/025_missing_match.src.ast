struct Point {
    x: int,
    y: int,
}

fn main() {
    let p0 = Point {
        x: 0,
        y: 0
    } in
    let Point { x: 10, y: 8 } = p0 in
    ()
}

