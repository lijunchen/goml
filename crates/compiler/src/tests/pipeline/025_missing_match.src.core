fn main() -> unit {
  let p0/0 = Point { x: 0, y: 0 } in
  let mtmp0 = p0/0 in
  let x1 = Point.x(mtmp0) in
  let x2 = Point.y(mtmp0) in
  match x2 {
    8 => {
      match x1 {
        10 => {
          ()
        },
        _ => missing(""),
      }
    },
    _ => missing(""),
  }
}