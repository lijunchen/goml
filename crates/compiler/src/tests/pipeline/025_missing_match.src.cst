FILE@0..126
  STRUCT@0..37
    StructKeyword@0..6 "struct"
    Whitespace@6..7 " "
    Uident@7..12 "Point"
    Whitespace@12..13 " "
    LBrace@13..14 "{"
    Whitespace@14..17 "\n  "
    STRUCT_FIELD_LIST@17..37
      STRUCT_FIELD@17..23
        Lident@17..18 "x"
        Colon@18..19 ":"
        Whitespace@19..20 " "
        TYPE_INT@20..23
          IntKeyword@20..23 "int"
      Comma@23..24 ","
      Whitespace@24..27 "\n  "
      STRUCT_FIELD@27..34
        Lident@27..28 "y"
        Colon@28..29 ":"
        Whitespace@29..30 " "
        TYPE_INT@30..34
          IntKeyword@30..33 "int"
          Whitespace@33..34 "\n"
      RBrace@34..35 "}"
      Whitespace@35..37 "\n\n"
  FN@37..126
    FnKeyword@37..39 "fn"
    Whitespace@39..40 " "
    Lident@40..44 "main"
    PARAM_LIST@44..47
      LParen@44..45 "("
      RParen@45..46 ")"
      Whitespace@46..47 " "
    BLOCK@47..126
      LBrace@47..48 "{"
      Whitespace@48..51 "\n  "
      EXPR_LET@51..124
        LetKeyword@51..54 "let"
        Whitespace@54..55 " "
        PATTERN_VARIABLE@55..58
          Lident@55..57 "p0"
          Whitespace@57..58 " "
        Eq@58..59 "="
        Whitespace@59..60 " "
        EXPR_LET_VALUE@60..81
          EXPR_STRUCT_LITERAL@60..81
            Uident@60..65 "Point"
            Whitespace@65..66 " "
            STRUCT_LITERAL_FIELD_LIST@66..81
              LBrace@66..67 "{"
              Whitespace@67..68 " "
              STRUCT_LITERAL_FIELD@68..72
                Lident@68..69 "x"
                Colon@69..70 ":"
                Whitespace@70..71 " "
                EXPR_INT@71..72
                  Int@71..72 "0"
              Comma@72..73 ","
              Whitespace@73..74 " "
              STRUCT_LITERAL_FIELD@74..79
                Lident@74..75 "y"
                Colon@75..76 ":"
                Whitespace@76..77 " "
                EXPR_INT@77..79
                  Int@77..78 "0"
                  Whitespace@78..79 " "
              RBrace@79..80 "}"
              Whitespace@80..81 " "
        InKeyword@81..83 "in"
        Whitespace@83..86 "\n  "
        EXPR_LET_BODY@86..124
          EXPR_LET@86..124
            LetKeyword@86..89 "let"
            Whitespace@89..90 " "
            PATTERN_CONSTR@90..111
              Uident@90..95 "Point"
              Whitespace@95..96 " "
              STRUCT_PATTERN_FIELD_LIST@96..111
                LBrace@96..97 "{"
                Whitespace@97..98 " "
                STRUCT_PATTERN_FIELD@98..103
                  Lident@98..99 "x"
                  Colon@99..100 ":"
                  Whitespace@100..101 " "
                  PATTERN_INT@101..103
                    Int@101..103 "10"
                Comma@103..104 ","
                Whitespace@104..105 " "
                STRUCT_PATTERN_FIELD@105..109
                  Lident@105..106 "y"
                  Colon@106..107 ":"
                  Whitespace@107..108 " "
                  PATTERN_INT@108..109
                    Int@108..109 "8"
                RBrace@109..110 "}"
                Whitespace@110..111 " "
            Eq@111..112 "="
            Whitespace@112..113 " "
            EXPR_LET_VALUE@113..116
              EXPR_LIDENT@113..116
                Lident@113..115 "p0"
                Whitespace@115..116 " "
            InKeyword@116..118 "in"
            Whitespace@118..121 "\n  "
            EXPR_LET_BODY@121..124
              EXPR_UNIT@121..124
                LParen@121..122 "("
                RParen@122..123 ")"
                Whitespace@123..124 "\n"
      RBrace@124..125 "}"
      Whitespace@125..126 "\n"