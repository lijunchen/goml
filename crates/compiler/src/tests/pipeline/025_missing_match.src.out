missing: 
panic: 

goroutine 1 [running]:
main.missing(...)
	${WORKDIR}/main.go:5
main.main0(...)
	${WORKDIR}/main.go:32
main.main()
	${WORKDIR}/main.go:38 +0x65
exit status 2
